core/src/data/value/error.rs

```diff
@@ -37,24 +37,16 @@ pub enum ValueError {
     #[error("failed to parse Decimal: {0}")]
     FailedToParseDecimal(String),
 
-    #[error("add on non-numeric values: {0:?} + {1:?}")]
-    AddOnNonNumeric(Value, Value),
-
-    #[error("subtract on non-numeric values: {0:?} - {1:?}")]
-    SubtractOnNonNumeric(Value, Value),
-
-    #[error("multiply on non-numeric values: {0:?} * {1:?}")]
-    MultiplyOnNonNumeric(Value, Value),
-
-    #[error("divide on non-numeric values: {0:?} / {1:?}")]
-    DivideOnNonNumeric(Value, Value),
+    #[error("non-numeric values {lhs:?} {operator:?} {rhs:?}")]
+    NonNumericMathOperation {
+        lhs: Value,
+        rhs: Value,
+        operator: String,
+    },
```

Comment:
Nice changes, thanks a lot 👍 

`operator: String` covers more than its actual range.
This `NonNumericMathOperation` is only for five numeric binary operations.
However, `String` is superset, does not have the identical with the previous state.

This was why @ever0de suggested to use enum for this.
```rust
pub enum Arithmetic {
    Plus,
    Minus,
    // ...
}
```

Let's have `NumericBinaryOperator` enum in `data/value/error.rs`
e.g.
```rust
// we can use strum_macros::Display in here
#[derive(Debug, PartialEq, Eq, Serialize, Display)]
pub enum NumericBinaryOperator {                   
    #[strum(to_string = "+")]                      
    Add,                                           
    #[strum(to_string = "-")]                      
    Subtract,                                      
    #[strum(to_string = "*")]                      
    Multiply,                                      
    #[strum(to_string = "/")]                      
    Divide,                                        
    #[strum(to_string = "%")]                      
    Modulo,                                        
}                                                  
```

Then,
```rust
#[error("non-numeric values {lhs:?} {operator} {rhs:?}")]
NonNumericMathOperation {                                
    lhs: Value,                                          
    rhs: Value,                                          
    operator: NumericBinaryOperator,                     
},                                                       
```
operator implements `Display` so you can replace `{operator:?}` to `{operator}`.

Reply:
enum value BinaryOperationOverflow  also have an operator with values of string.  Should I go ahead and change this to use NumericBinaryOperator?

Reply:
you are right I have an enum error using exactly char type.

Reply:
Ok.  I will also make that change was well to remove char/string from enum errors.

Reply:
Done!!